### PR TITLE
feature/pfcron-tests

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Tests for pfcron

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Improved tests for pfcron jobs

### CODE REVIEW

**Comment by julsemaan** [comment]
The current tests look like what we need to properly unit test the tasks

Make sure you create a test for each of the items below (except Fingerbank data update):
image

Some of them (Golang based) are still missing from this PR, my opinion would be to focus on them before the Perl ones that aren't in that list above

File: go/cron/window_batch_task_test.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+				additional_args,
+			),
+		),
+		setupSql,
+		tests,
+		cleanupSQL,
+	)
+}
+
+func TestAdminApiAuditLog(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/window_batch_task_test.go]:
```suggestion
func TestAdminApiAuditLogCleanup(t *testing.T) {
```

File: go/cron/window_batch_task_test.go [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+				name:          "admin_api_audit_log entries left",
+				sql:           ` SELECT COUNT(*) FROM admin_api_audit_log `,
+				expectedCount: 8,
+			},
+		},
+		[]string{"DELETE FROM admin_api_audit_log"},
+	)
+}
+
+func TestAuthLog(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/window_batch_task_test.go]:
```suggestion
func TestAuthLogCleanup(t *testing.T) {
```

File: go/cron/window_batch_task_test.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+				name:          "auth_log entries left",
+				sql:           ` SELECT COUNT(*) FROM auth_log `,
+				expectedCount: 8,
+			},
+		},
+		[]string{"DELETE FROM auth_log"},
+	)
+}
+
+func TestDnsAuditLog(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/window_batch_task_test.go]:
```suggestion
func TestDnsAuditLogCleanup(t *testing.T) {
```

File: go/cron/ip4log_cleanup_test.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
+package maint
+
+import (
+	"testing"
+)
+
+func TestIp4logNoRotate(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/ip4log_cleanup_test.go]:
```suggestion
func TestIp4logCleanupNoRotate(t *testing.T) {
```

File: go/cron/ip4log_cleanup_test.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                `,
+				expectedCount: 8,
+			},
+		},
+		[]string{"DELETE FROM ip4log_history"},
+	)
+
+}
+
+func TestIp4logRotate(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/ip4log_cleanup_test.go]:
```suggestion
func TestIp4logCleanupRotate(t *testing.T) {
```

File: go/cron/ip6log_cleanup_test.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
+package maint
+
+import (
+	"testing"
+)
+
+func TestIp6logNoRotate(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/ip6log_cleanup_test.go]:
```suggestion
func TestIp6logCleanupNoRotate(t *testing.T) {
```

File: go/cron/ip6log_cleanup_test.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+				sql:           "SELECT COUNT(*) FROM ip6log_history",
+				expectedCount: 8,
+			},
+		},
+		[]string{"DELETE FROM ip6log_history"},
+	)
+
+}
+
+func TestIp6logRotate(t *testing.T) {
```
Comment (julsemaan) [inline go/cron/ip6log_cleanup_test.go]:
```suggestion
func TestIp6logCleanupRotate(t *testing.T) {
```

**Comment by extrafu** [comment]
@julsemaan Have a look and if things look good, we could merge that sooner than later so @nqb could fix any issues related to Venom if any.

**Comment by julsemaan** [comment]
Just need @jrouzierinverse to address the small details I posted in my review and we're good to merge

**Comment by nqb** [comment]
@jrouzierinverse,

- Please add your test suite in a `run_tests.yml` inside a scenario (for example `unit_tests`) in order to run it.
- Could you use a full path to `pfcron-test.log` everywhere in the test suite ?

EDIT: rename `.yaml` by `.yml`

**Comment by nqb** [comment]
I did adjustments mentioned above and run a pipeline:

https://gitlab.com/inverse-inc/packetfence/-/pipelines/397235188

It pipeline passed, I'm ok to merge.

**Comment by nqb** [comment]
New pipeline: https://gitlab.com/inverse-inc/packetfence/-/pipelines/397389956

**Comment by nqb** [comment]
Pipeline failed for:
- Perl unit tests
- pfcron integration tests

⚠️ I ran this pipeline **before** @julsemaan's commits ⚠️ 

### Perl unit tests

```
Oct 28 11:12:58.759ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] eth1.1: error fetching interface information: Device not found
Oct 28 11:12:59.088ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] eth1.2: error fetching interface information: Device not found
Oct 28 11:12:59.506ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] 
Oct 28 11:12:59.506ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] #   Failed test 'security event opened'
Oct 28 11:12:59.506ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] #   at unittest/node-maintenance.t line 74.
Oct 28 11:12:59.506ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] # Looks like you planned 3 tests but ran 8.
Oct 28 11:12:59.506ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] # Looks like you failed 1 test of 8 run.
Oct 28 11:13:00.265ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] unittest/node-maintenance.t ........................................
Oct 28 11:13:00.265ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] Dubious, test returned 1 (wstat 256, 0x100)
Oct 28 11:13:00.265ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] All 3 subtests passed
[..]
Oct 28 11:23:18.017ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] Can't locate Net/SMTP/Server.pm in @INC (you may need to install the Net::SMTP::Server module) (@INC contains: /usr/local/pf/lib /usr/local/pf/lib_perl/lib/perl
5/x86_64-linux-thread-multi /usr/local/pf/lib_perl/lib/perl5 /usr/local/pf/t /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5) at unittest/pfcron/task.t line 47.
Oct 28 11:23:18.017ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] BEGIN failed--compilation aborted at unittest/pfcron/task.t line 47.
Oct 28 11:23:18.017ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] # Looks like your test exited with 2 just after 1.
Oct 28 11:23:18.280ESC[37m [DEBU] [Perl tests] [run_perl_tests] [exec] unittest/pfcron/task.t .............................................
```

**Comment by nqb** [comment]
### Venom tests

The file `/usr/local/pf/t/venom/test_suites/pfcron/pfcron-test.log` is not created by the `pfcron` logger.

**Comment by jrouzierinverse** [comment]
readfile does not respect absolute paths.
That is why it is failing.

I will look into the perl test failing.

**Pull request merged**